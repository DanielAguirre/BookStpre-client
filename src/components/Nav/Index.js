import React from 'react';
import { Link } from 'react-router-dom';
import glamorous from 'glamorous'
import Login from '../Login';

const Nav = glamorous.nav({
  position: 'absolute',
  color: 'white',
  width: '100%'
});

const NavContainer = glamorous.ul({
  display: 'flex',
  'flexDirection': 'row',
  'justifyContent': 'center',
  'alignContent': 'center'
});

 const Navigation = () => (
  <Nav className="navigation">
    <NavContainer className="navigation-container">
      <Link to="/" className="navigation-element">Home</Link>
      <Link to="/add" className="navigation-element">Add Book</Link>
      <Link to="/user" className="navigation-element">User</Link>
      <Login />
    </NavContainer>
  </Nav>
);

module.exports =  Navigation;