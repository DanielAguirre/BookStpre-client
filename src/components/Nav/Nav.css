.navigation-element{
  display: inline-block;
  width: 200px;
  font-size: 1.5rem;
  text-align: center;
  color: inherit;
  text-decoration: none;
}