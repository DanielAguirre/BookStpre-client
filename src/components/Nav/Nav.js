import React from 'react';
import './Nav.css';

const Nav = () => (
  <nav className="navigation">
    <ul className="navigation-container">
      <li>
        <a href="/">Home </a>
      </li>
      <li>
        <a href="/add">Add Book </a>
      </li>
      <li>
        <a href="">User </a>
      </li>
    </ul>
  </nav>
);

module.exports = Nav;