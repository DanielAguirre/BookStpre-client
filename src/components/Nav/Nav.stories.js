import React from 'react';
import { MemoryRouter } from 'react-router';
import { storiesOf, action, linkTo, setAddon } from '@kadira/storybook';
import infoAddon from '@kadira/react-storybook-addon-info';
import StoryRouter from 'storybook-router';

import Nav from './Nav';

setAddon(infoAddon)

storiesOf('Navigation', module)
  .addDecorator(StoryRouter())
  .addDecorator(story => (
    <MemoryRouter initialEntries={['/']}>{story()}</MemoryRouter>
  ))
  .addWithInfo('Navigation', 'description', () => (
    <Nav onClick={action('clicked')}/>
  ));

