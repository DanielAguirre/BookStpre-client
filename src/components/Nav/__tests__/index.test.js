import React from 'react'
import { shallow } from 'enzyme';
import Nav from '../index';

it('<Nav />', ()=> {
  const component = shallow(<Nav/>)
  expect(component).toMatchSnapshot();
})