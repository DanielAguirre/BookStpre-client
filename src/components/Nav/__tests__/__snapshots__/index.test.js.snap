exports[`test <Nav /> 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <glamorous(nav)
    className="navigation">
    <glamorous(ul)
        className="navigation-container">
        <Link
            className="navigation-element"
            replace={false}
            to="/">
            Home
        </Link>
        <Link
            className="navigation-element"
            replace={false}
            to="/add">
            Add Book
        </Link>
        <Link
            className="navigation-element"
            replace={false}
            to="/user">
            User
        </Link>
        <Login />
    </glamorous(ul)>
</glamorous(nav)>,
  "nodes": Array [
    <glamorous(nav)
      className="navigation">
      <glamorous(ul)
            className="navigation-container">
            <Link
                  className="navigation-element"
                  replace={false}
                  to="/">
                  Home
            </Link>
            <Link
                  className="navigation-element"
                  replace={false}
                  to="/add">
                  Add Book
            </Link>
            <Link
                  className="navigation-element"
                  replace={false}
                  to="/user">
                  User
            </Link>
            <Login />
      </glamorous(ul)>
</glamorous(nav)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Navigation />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <glamorous(nav)
          className="navigation">
          <glamorous(ul)
                    className="navigation-container">
                    <Link
                              className="navigation-element"
                              replace={false}
                              to="/">
                              Home
                    </Link>
                    <Link
                              className="navigation-element"
                              replace={false}
                              to="/add">
                              Add Book
                    </Link>
                    <Link
                              className="navigation-element"
                              replace={false}
                              to="/user">
                              User
                    </Link>
                    <Login />
          </glamorous(ul)>
</glamorous(nav)>,
        "_debugID": 2,
        "_renderedOutput": <glamorous(nav)
          className="navigation">
          <glamorous(ul)
                    className="navigation-container">
                    <Link
                              className="navigation-element"
                              replace={false}
                              to="/">
                              Home
                    </Link>
                    <Link
                              className="navigation-element"
                              replace={false}
                              to="/add">
                              Add Book
                    </Link>
                    <Link
                              className="navigation-element"
                              replace={false}
                              to="/user">
                              User
                    </Link>
                    <Login />
          </glamorous(ul)>
</glamorous(nav)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Navigation />,
}
`;
